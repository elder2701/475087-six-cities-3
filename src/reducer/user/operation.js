import {ActionCreator, AuthorizationStatus} from "./user.js";

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
        .then((res) => {
          dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
          dispatch(ActionCreator.loadUserInfo(res.data));
        })
        .catch((err) => {
          throw err;
        });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
    .then((res) => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.loadUserInfo(res.data));
    });
  },
};

export {Operation};
