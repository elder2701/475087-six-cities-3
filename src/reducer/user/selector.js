import NameSpace from "../name-space.js";

const getAuthStatus = (state)=> state[NameSpace.USER].authorizationStatus;

export {getAuthStatus};
