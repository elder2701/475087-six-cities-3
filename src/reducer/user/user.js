import {extend} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.AUTH,
  userInfo: {}
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  LOAD_USER_INFO: `LOAD_USER_INFO`
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  loadUserInfo: (userInfo) => {
    return {
      type: ActionType.LOAD_USER_INFO,
      payload: userInfo
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.LOAD_USER_INFO:
      return extend(state, {
        userInfo: action.payload
      });
  }
  return state;
};

export {ActionCreator, reducer, AuthorizationStatus};
