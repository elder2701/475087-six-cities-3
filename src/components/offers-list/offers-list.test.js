import React from "react";
import renderer from "react-test-renderer";
import OffersList from "./offers-list.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import history from "../../history.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

const offers = [
  {
    city: {
      name: ``,
      location: {latitude: 1, longitude: 1, zoom: 1}
    },
    id: 12,
    previewImage: ``,
    price: 8220,
    rating: 3,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isFavorite: false,
    images: [``, ``],
    bedrooms: 1,
    goods: [``, ``],
    description: ``,
    location: {
      latitude: 1,
      longitude: 2,
      zoom: 3
    },
    isPremium: true,
    maxAdults: 1,
    hostId: 1,
    hostName: ``,
    hostIsPro: true,
    hostAvatarUrl: ``
  },
  {
    city: {
      name: ``,
      location: {latitude: 1, longitude: 1, zoom: 1}
    },
    id: 1,
    previewImage: ``,
    price: 8220,
    rating: 3,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isFavorite: false,
    images: [``, ``],
    bedrooms: 1,
    goods: [``, ``],
    description: ``,
    location: {
      latitude: 2,
      longitude: 2,
      zoom: 3
    },
    isPremium: true,
    maxAdults: 1,
    hostId: 3,
    hostName: ``,
    hostIsPro: true,
    hostAvatarUrl: ``
  }
];

it(`<OffersList /> sould be render`, () => {
  const store = mockStore({[NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
  }});
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <OffersList
              cityOffers={offers}
              type={``}
              onUpdateStatus={() => {}}
              onHoverActiveCard={() => {}}
            />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
