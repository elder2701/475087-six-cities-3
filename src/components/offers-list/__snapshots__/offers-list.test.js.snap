// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersList /> sould be render 1`] = `
<div
  className=" places__list"
>
  <article
    className="cities__place-card place-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="cities__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            8220
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        />
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="cities__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            101320
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/10"
          onClick={[Function]}
        />
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="cities__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            83210
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        />
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="cities__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            8210
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        />
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
